include/engine/datafacade/internal_datafacade.hpp

```diff
@@ -742,6 +775,35 @@ class InternalDataFacade final : public BaseDataFacade
         }
     }
 
+    // Returns the data source ids that were used to supply the edge
+    // weights.
+    virtual void
+    GetUncompressedReverseDatasources(const EdgeID id,
+                                      std::vector<uint8_t> &result_datasources) const override final
+    {
+        const unsigned begin = m_geometry_indices.at(id);
+        const unsigned end = m_geometry_indices.at(id + 1) - 1;
+
+        result_datasources.clear();
+        result_datasources.reserve(end - begin);
```

Comment:
since we clear and allocate anyhow, we should really move on to returning a moved out vector.
In theory I would prefer so simply have an iterator view of this, so we don't copy back and forward.